src/auth/dto/authcreds.dto.ts

```diff
@@ -0,0 +1,15 @@
+import { IsString, IsNotEmpty, IsEmail } from 'class-validator';
+import { ApiProperty } from '@nestjs/swagger';
+
+export class AuthCredsDto {
+  @IsString()
+  @ApiProperty({ description: 'The email of the user' })
+  @IsEmail()
```

Comment:
PullProofAI

There is a potential issue with the validation logic for the email field. The `@IsEmail()` decorator is used along with `@IsString()` and `@IsNotEmpty()`. This combination might lead to unexpected behavior as `@IsEmail()` already checks for a valid email format which includes being a non-empty string. Consider removing `@IsString()` and `@IsNotEmpty()` decorators for the email field to avoid redundant checks.

```diff
-  @IsString()
-  @IsNotEmpty()
   @ApiProperty({ description: 'The email of the user' })
   @IsEmail()
   email: string;
```


<!-- This is an auto-generated comment by PullProofAI -->